src/app/components/HomePage/header.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import styles from './header.scss';
+import LoginAndSignup from './loginAndSignup';
+import Username from './username';
+
+function Header({ uname }) {
+  return (
+    <div className={styles.header}>
+      <div className={styles.nav}>
+        <div className={styles.logo}>
+          <Link to="/">
+            <img src="./../public/images/logo.png" alt="logo" />
+          </Link>
+        </div>
+        <div className={styles.menu}>
+          <ul>
+            <li>
+              <Link to="/info" style={{ textDecoration: 'none', color: 'rgba(0, 0, 0, 0.5)' }}>
+                Thông tin
+              </Link>
+            </li>
+          </ul>
+        </div>
+        <div className={styles.user}>{uname === '' ? <LoginAndSignup /> : <Username />}</div>
+      </div>
+    </div>
+  );
+}
+
+Header.propTypes = {
+  uname: PropTypes.string.isRequired,
+};
+
+export default Header;
```

Comment:
This header file will be used across many pages, so let create a folder name `NavBar` that has the same level with `Page...` level
```
app/
|__components/
    |__ AboutPage/
    |__ HomePage/
    |__ PageNotFound/
    |__ Navbar/
```

This structure is not ideal but lets go with it first